_posts/2020-06-26-Making-IrishGen-SPARQL-Part-I.md

```diff
@@ -0,0 +1,549 @@
+children a person had would occasionally be recorded.  Thus we can ask
+the question who had the most recorded children in the genealogies:
+
+```sparql
+prefix irishRel: <http://example.com/earlyIrishRelationship.ttl#> 
+
+select ?x (max(?numChild) as ?maxNumChild) 
+from <tag:stardog:api:context:all>
+where {
+    ?x irishRel:numChild ?numChild
+} 
+group by ?x
+order by desc (?maxNumChild)
+```
+
+This query takes all URLs that have a `irishRel:numChild` declared on
+them then counts the `max` of them for each individual.  This means
+that it will find the largest `irishGen:numChild` that an individual
+has declared on them.  It orders them _descending_ (desc) and it
+[groups them
+by](https://www.w3.org/TR/2013/REC-sparql11-query-20130321/#groupby)
+`?x` which means that the query will break each group represented by
+the variable `?x` into a separate group then work on that group.  In
+this case, it is `?x` so that each URL is counted as its own group.
+The result is an order list going from most to least which shows each
+individual and their `:numChild` which is more than 1000 results.
+Thankfully, this is a very fast query at 68 milliseconds.
+Benchmarking software is a exhaustingly large subject so there is a
+very high variance in what the user may experience on their own
+computers due to _inter alia_ CPU L1 and L2 cache strategies, RAM
+clock rate, disk speed, query optimiser, and even more complicated
+events outside the control of the Triplestore.
+
+As the result is over 1000, we will only concern ourselves with the
+top five. We can do that progrmmatically by using the `limit` keyword.
+
+```sparql
+prefix irishRel: <http://example.com/earlyIrishRelationship.ttl#> 
+
+select ?x (max(?numChild) as ?maxNumChild) 
+from <tag:stardog:api:context:all>
+where {
+    ?x irishRel:numChild ?numChild
+} 
+group by ?x
+order by desc (?maxNumChild)
+limit 5
+```
+
+
+| ?x                                                                                                      | ?maxNumChild |
+| http://example.com/Laud_Misc_610/CGH/do_minigud_senchais_fer_muman.trig#Óengus                          | 48           |
+| http://example.com/Rawl_B502/de_genelogia_síl_ébir_16.trig#Óengus-e74701aa                              | 48           |
+| http://example.com/Rawl_B502/de_peritia_&_de_genealogis_dál_niad_cuirp_incipit.trig#CathairMár          | 33           |
+| http://example.com/Rawl_B502/de_peritia_&_de_genealogis_dál_niad_cuirp_incipit.trig#CathairMár-82bed530 | 30           |
+| http://example.com/LL/de_genelach_dáil_nia_corbb.trig#CathairMár                                        | 30           |
+
+The results of this raise an interesting point.  The reader will
+notice that Cathair Már appears three times in the results.  This
+means that there are _three_ different declared number of children for
+Cathair Már and they are not counted the same as `max` should return
+only the largest for a single individual.  To investigate this is
```

Comment:
Perhaps explain why the query returns results for individuals as opposed to instances of individuals.